File: planning-idsdp-rms-frontend/src/app/main/modules/rpm/models/ResearcherPresentation.ts
import {ResearcherProposal} from "./ResearcherProposal";
import {CreateSeminar} from "./CreateSeminar";

export class ResearcherPresentation {

    id: number;
    uuid: string;
    m2CreateSeminarId: number;
    m1ResearcherProposalId: number;
    presentationType: string;
    presentationStatus: string;
    createSeminarDto: CreateSeminar;
    researcherProposalDto: ResearcherProposal;
}

File: planning-idsdp-rms-rpm-backend/src/main/java/com/ibcs/idsdp/rpm/model/domain/ResearcherPresentation.java
package com.ibcs.idsdp.rpm.model.domain;

import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.OneToOne;
import javax.persistence.Table;

import com.ibcs.idsdp.common.model.domain.BaseEntity;
import com.sun.istack.NotNull;

import lombok.Data;

@Data
@Entity
@Table(name = "m2_researcher_presentation")
public class ResearcherPresentation extends BaseEntity {
	
	@NotNull
	@OneToOne
	@JoinColumn(name = "m2_create_seminar_id")
	private CreateSeminar createSeminar;
	
	@NotNull
	@OneToOne
	@JoinColumn(name = "m1_researcher_proposal_id")
    private ResearcherProposal researcherProposal;
	
	@NotNull
	private Long presentationStatus;

	@NotNull
	private Long presentationType;
	
	private Boolean isEditable;

}
